### PR TITLE
Fix add material for ensembles

### PR DESCRIPTION
**Pull request by stvdwtt**
Address #220 

- Element activation now uses the `dof_handler` for each ensemble member instead of sharing the zeroth member's
- Added a test for data assimilation with material addition

### CODE REVIEW

File: application/adamantine.hh [hunk 1]
```diff
@@ -1601,10 +1601,17 @@ run_ensemble(MPI_Comm const &communicator,
...
-
-  auto elements_to_activate = adamantine::get_elements_to_activate(
-      thermal_physics_ensemble[0]->get_dof_handler(),
-      material_deposition_boxes);
+  std::vector<std::vector<
+      std::vector<typename dealii::DoFHandler<dim>::active_cell_iterator>>>
+      elements_to_activate_ensemble;
+  for (unsigned int member = 0; member < ensemble_size; ++member)
+  {
+    auto elements_to_activate = adamantine::get_elements_to_activate(
```
Comment (Rombur) [inline application/adamantine.hh:1609]:
```suggestion
    elements_to_activate_ensemble[member] = adamantine::get_elements_to_activate(
```

File: application/adamantine.hh [hunk 2]
```diff
@@ -1649,15 +1656,18 @@ run_ensemble(MPI_Comm const &communicator,
...
       timers[adamantine::add_material_search].start();
-      elements_to_activate = adamantine::get_elements_to_activate(
-          thermal_physics_ensemble[0]->get_dof_handler(),
-          material_deposition_boxes);
-      timers[adamantine::add_material_search].stop();
-
+      elements_to_activate_ensemble.clear();
       for (unsigned int member = 0; member < ensemble_size; ++member)
       {
+        auto elements_to_activate = adamantine::get_elements_to_activate(
```
Comment (Rombur) [inline application/adamantine.hh:1662]:
```suggestion
        auto elements_to_activate_ensemble[member] = adamantine::get_elements_to_activate(
```

File: tests/test_integration_da.cc [hunk 3]
```diff
@@ -55,3 +55,57 @@ BOOST_AUTO_TEST_CASE(integration_data_assimilation)
...
+{
+  /*
+   * This integration test checks that the data assimilation using point cloud
+   * data works while adding material.
+   *
+   * TODO: Add a similar test for ray data.
+   */
+  std::cout
+      << "Starting test case: integration_3D_da_point_cloud_add_material..."
+      << std::endl;
```
Comment (Rombur) [inline tests/test_integration_da.cc:69]: Please remove this. It's useless in the CI. There are environment variables for Boost Test that will do this for you.

File: application/adamantine.hh [hunk 4]
```diff
@@ -1601,10 +1601,17 @@ run_ensemble(MPI_Comm const &communicator,
...
+      elements_to_activate_ensemble(ensemble_size);
+  for (unsigned int member = 0; member < ensemble_size; ++member)
+  {
+    elements_to_activate_ensemble[member] =
+        adamantine::get_elements_to_activate(
+            thermal_physics_ensemble[member]->get_dof_handler(),
+            material_deposition_boxes);
+    timers[adamantine::add_material_search].stop();
+  }
   timers[adamantine::add_material_search].stop();
```
Comment (Rombur) [inline application/adamantine.hh:1615]: You forgot to remove the timers two lines above

**Review by Rombur — COMMENTED** [review]
Looks fine except for the `timers` issue

**Comment by stvdwtt** [comment]
[Updated CI](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/adamantine/detail/adamantine/368/pipeline)

**Review by Rombur — APPROVED** [review]

**Pull request merged**